web/views.py

```
@@ -4876,19 +4900,48 @@ def send_donation_thank_you_email(donation):
-        except Donation.DoesNotExist:
-            pass
+        else:
+            messages.error(request, "Donation has not been successfully processed.")
+            return redirect("donate")
 
-    # If no valid donation ID, redirect to the donate page
-    return redirect("donate")
+    except Donation.DoesNotExist:
+        messages.error(request, "Invalid donation ID.")
+        return redirect("donate")
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)